internal/controller/biosversion_controller_test.go

```diff
@@ -0,0 +1,426 @@
+
+		By("Ensuring that the Maintenance resource has been referenced by biosVersion")
+		Eventually(Object(biosVersion)).Should(SatisfyAny(
+			HaveField("Spec.ServerMaintenanceRef", &v1.ObjectReference{
+				Kind:       "ServerMaintenance",
+				Name:       serverMaintenance.Name,
+				Namespace:  serverMaintenance.Namespace,
+				UID:        serverMaintenance.UID,
+				APIVersion: serverMaintenance.GroupVersionKind().GroupVersion().String(),
+			}),
+			HaveField("Spec.ServerMaintenanceRef", &v1.ObjectReference{
+				Kind:       "ServerMaintenance",
+				Name:       serverMaintenance.Name,
+				Namespace:  serverMaintenance.Namespace,
+				UID:        serverMaintenance.UID,
+				APIVersion: "metal.ironcore.dev/v1alpha1",
+			}),
+		))
+
+		By("Ensuring that the biosVersion has Inprogress state and waiting")
+		Eventually(Object(biosVersion)).Should(
+			HaveField("Status.State", metalv1alpha1.BIOSVersionStateInProgress),
+		)
+
+		By("Approving the maintenance")
+		Eventually(Update(serverClaim, func() {
+			metautils.SetAnnotation(serverClaim, metalv1alpha1.ServerMaintenanceApprovalKey, "true")
+		})).Should(Succeed())
+
+		By("Ensuring that Server in Maintenance state")
+		Eventually(Object(server)).Should(SatisfyAll(
+			HaveField("Status.State", metalv1alpha1.ServerStateMaintenance),
+			HaveField("Spec.ServerMaintenanceRef", &v1.ObjectReference{
+				Kind:       "ServerMaintenance",
+				Name:       serverMaintenance.Name,
+				Namespace:  serverMaintenance.Namespace,
+				UID:        serverMaintenance.UID,
+				APIVersion: "metal.ironcore.dev/v1alpha1",
+			}),
+		))
+
+		ensureBiosVersionConditionTransisition(ctx, acc, biosVersion, server)
+
+		By("Ensuring that BIOS upgrade has completed")
+		Eventually(Object(biosVersion)).Should(
+			HaveField("Status.State", metalv1alpha1.BIOSVersionStateCompleted),
+		)
+
+		By("Ensuring that BIOSVersion has removed Maintenance")
+		Eventually(Object(biosVersion)).Should(
+			HaveField("Spec.ServerMaintenanceRef", BeNil()),
+		)
+		Consistently(Object(biosVersion)).Should(
+			HaveField("Spec.ServerMaintenanceRef", BeNil()),
+		)
+
+		Consistently(ObjectList(&serverMaintenanceList)).Should(HaveField("Items", BeEmpty()))
+
+		By("Deleting the BIOSVersion")
+		Expect(k8sClient.Delete(ctx, biosVersion)).To(Succeed())
+
+		By("Ensuring that the BiosVersion has been removed")
+		Eventually(Get(biosVersion)).Should(Satisfy(apierrors.IsNotFound))
+		Consistently(Get(biosVersion)).Should(Satisfy(apierrors.IsNotFound))
+	})
+})
+
+func ensureBiosVersionConditionTransisition(
+	ctx context.Context,
+	acc *conditionutils.Accessor,
+	biosVersion *metalv1alpha1.BIOSVersion,
+	server *metalv1alpha1.Server,
+) {
+	By("Ensuring that BIOS Conditions have reached expected state 'biosVersionUpgradeIssued'")
+	condIssue := &metav1.Condition{}
+	Eventually(
+		func(g Gomega) int {
+			g.Expect(k8sClient.Get(ctx, client.ObjectKey{Name: biosVersion.Name}, biosVersion)).To(Succeed())
```

Comment:
Can we use here `komega.Get()` instead of `client.Get` this will be then consitent throughout the test suite.